### PR TITLE
Ports Crew Objectives from Oracle

### PR DESCRIPTION
**Pull request by Alex-Selvey**
Original port from ike709

ports crew objectives for each crewmember (except special roles)
https://github.com/Interstation13/Interstation13/pull/15

[OracleStation/OracleStation#626](https://github.com/OracleStation/OracleStation/pull/626)
[OracleStation/OracleStation#708](https://github.com/OracleStation/OracleStation/pull/708)

:cl: Orginal code ike709 ported by MemeProof
add: Crew Objectives
server: Added global and preferences option for crew objectives
/:cl:

### CODE REVIEW

File: code/__HELPERS/roundend.dm [hunk 1]
```diff
@@ -217,7 +217,22 @@
 	for(var/antag_name in total_antagonists)
 		var/list/L = total_antagonists[antag_name]
 		log_game("[antag_name]s :[L.Join(", ")].")
+	
+	CHECK_TICK
+
+	// Sunset - added to display if own personal objectives were a success
```
Comment (ghost) [inline code/__HELPERS/roundend.dm]: `sunset start` and `sunset end`

File: code/controllers/subsystem/ticker.dm [hunk 2]
```diff
@@ -57,6 +57,11 @@ SUBSYSTEM_DEF(ticker)
 	var/mode_result = "undefined"
 	var/end_state = "undefined"
 
+	//Sunset - Crew Objectives
```
Comment (ghost) [inline code/controllers/subsystem/ticker.dm]: can be modularized in `sunset/code/controllers/subsystem/ticker.dm`

like
```
/datum/controller/subsystem/ticker
	var/list/successfulCrew = list()
	var/list/crewobjlist = list()
	var/list/crewobjjobs = list()
```

File: code/controllers/subsystem/ticker.dm [hunk 3]
```diff
@@ -124,6 +129,14 @@ SUBSYSTEM_DEF(ticker)
 		gametime_offset = rand(0, 23) HOURS
 	else if(CONFIG_GET(flag/shift_time_realtime))
 		gametime_offset = world.timeofday
+
```
Comment (ghost) [inline code/controllers/subsystem/ticker.dm]: `sunset start` and `sunset end`

**Review by ghost — CHANGES_REQUESTED** [review]
use tabs, not spaces, to indent

File: code/__HELPERS/roundend.dm (hunk 1 above)
Comment (Alex-Selvey) [reply code/__HELPERS/roundend.dm]: I have updated the comments.

File: code/controllers/subsystem/ticker.dm (hunk 2 above)
Comment (Alex-Selvey) [reply code/controllers/subsystem/ticker.dm]: I have created ticker under sunset and have modularized this portion. Thank you for catching that.

File: code/controllers/subsystem/ticker.dm (hunk 3 above)
Comment (Alex-Selvey) [reply code/controllers/subsystem/ticker.dm]: I have updated the comments.

**Comment by SunsetStationuser** [comment]
@MemeProof  has this been tested?

**Comment by Alex-Selvey** [comment]
I was told something in tgcode broke this, but was fixed. This needs to be retested. I'll cue it for my own testing this weekend.

**Comment by Alex-Selvey** [comment]
Just checked and is still confirmed to not be working after whatever was changed in tgcode. Need help reviewing what is causing the issue.

**Comment by Alex-Selvey** [comment]
I take it back, Crew objectives show up in chat just fine. This can be merged

**Comment by Alex-Selvey** [comment]
image

**Pull request merged**